tests/primeqa/mrc/processors/preprocessors/test_tydiqa_bpes_preprocessor.py

```diff
@@ -0,0 +1,56 @@
+import datasets
+import pytest
+from datasets import Dataset
+from transformers import AutoTokenizer
+
+from primeqa.mrc.processors.preprocessors.tydiboolqa_bpes import TyDiBoolQAPreprocessor
+from tests.primeqa.mrc.common.base import UnitTest
+import numpy as np
+
+class TestTyDiBoolQAPreprocessor(UnitTest):
+
+    @pytest.mark.flaky(reruns=10, reruns_delay=30)  # Account for intermittent S3 errors downloading HF data
+    @pytest.fixture(scope='session')
+    def train_examples(self):
+        examples = datasets.load_dataset("tydiqa", "primary_task", split='train[:100]')
+        return examples
+
+    @pytest.mark.flaky(reruns=10, reruns_delay=30)  # Account for intermittent S3 errors downloading HF data
+    @pytest.fixture(scope='session')
+    def eval_examples(self):
+        examples = datasets.load_dataset("tydiqa", "primary_task", split='validation[:100]')
+        return examples
+
+    @pytest.fixture(scope='class')
+    def tokenizer(self):
+        return AutoTokenizer.from_pretrained('xlm-roberta-base')
+
+    @pytest.fixture(scope='class')
+    def tydiqa_preprocessor(self, tokenizer):
+        return TyDiBoolQAPreprocessor(
+            tokenizer,
+            stride=128,
+            load_from_cache_file=False,
+        )
+
+    def test_train_preprocessing_runs_without_errors(self, train_examples, tydiqa_preprocessor):
+        train_examples, train_features = tydiqa_preprocessor.process_train(train_examples)
```

Comment:
Should there be some checks to make sure nothing else other than for Y/N has been updated?

Reply:
I am OK with this test case